migrate/list-convert.go

```diff
@@ -0,0 +1,296 @@
+
+		// The name of the new drive should be its FSUUID
+		if NewDrive.ObjectMeta.Name == NewDrive.Status.FSUUID {
+			fmt.Println("Validate that the name of the new drive is its FSUUID")
+			// We need to also add validation to check if this is unique among the entire drive list set for conversion.
+			// https://stackoverflow.com/questions/9251234/how-to-check-the-uniqueness-inside-a-for-loop
+			if _, ok := setOfUniqueFSUUIDs[NewDrive.Status.FSUUID]; ok {
+				fmt.Println("FSUUID found already, we can't convert this drive unless is unique: ", NewDrive.Status.FSUUID)
+			} else {
+				fmt.Println("element not found, FSUUID is unique, proceed to be added/appended for convertion")
+				setOfUniqueFSUUIDs[NewDrive.Status.FSUUID] = struct{}{}
+				// Append the new drive to the list
+				newDrives = append(newDrives, NewDrive)
+			}
+		}
+	}
+
+	// Return the new list of drives
+	return newDrives, nil
+}
+
+func main() {
+
+	// Get the kubernetes configuration file
+	kubeConfig, err := getKubeConfig()
+	if err != nil {
+		fmt.Printf("%s: Could not connect to kubernetes. %s=%s\n", "Error", "KUBECONFIG", kubeConfig)
+		os.Exit(1)
+	}
+
+	// Get the latest interface available
+	driveInterface, err := directcsiclient.DirectCSIDriveInterfaceForConfig(kubeConfig)
+	if err != nil {
+		fmt.Printf("%s: could not initialize drive adapter client: err=%v\n", "Error", err)
+		os.Exit(1)
+	}
+
+	// Get volume interface
+	volumeInterface, err := directcsiclient.DirectCSIVolumeInterfaceForConfig(kubeConfig)
+	if err != nil {
+		fmt.Printf("%s: could not initialize drive adapter client: err=%v\n", "Error", err)
+		os.Exit(1)
+	}
+
+	// Then we list the drives
+	// This is getting all the info from all the drives
+	ctx, _ := context.WithCancel(context.Background())
+	driveList, err := driveInterface.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		fmt.Println("error")
+		os.Exit(1)
+	}
+	volumeList, err := volumeInterface.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		fmt.Println("error")
+		os.Exit(1)
+	}
+
+	// Create volume list
+	filteredVolumes := []directcsi.DirectCSIVolume{}
+	for _, volume := range volumeList.Items {
+		// We currently don't filter volumes
+		filteredVolumes = append(filteredVolumes, volume)
+	}
+
+	// To filter the drives by status and version:
+	filteredDrives := []directcsi.DirectCSIDrive{}
+	for _, drive := range driveList.Items {
+		isDriveReadyOrInUse := false
+		isDrivev1beta5Version := false
+		isFilesystemUUIDPopulated := false
+		isPathInDevNotation := false
+		isDriveFinalizerCorrect := false
+		// Only for Ready and InUse status are valid here for this list of migration:
+		if drive.Status.DriveStatus == directcsi.DriveStatusReady || drive.Status.DriveStatus == directcsi.DriveStatusInUse {
+			isDriveReadyOrInUse = true
+		}
+		// Only for version v1beta5 is applicable the migration
+		groupAndVersion := strings.Split(drive.TypeMeta.APIVersion, `/`)
+		gvk, _ := directcsiclient.GetGroupKindVersions(
```

Comment:
We don't need to run discovery for each drive, Discovery is based on Group and Kind. We can move this out of the loop